cmd/proc_windows.go

```diff
@@ -21,9 +21,16 @@ import (
 )
 
 func gracefullyStopProcess(pid int) error {
-	cmd := exec.Command("taskkill", "/pid", strconv.Itoa(pid))
+	fmt.Printf("Stop...")
+	// process on windows will not stop unless forced with /f
+	cmd := exec.Command("taskkill", "/pid", strconv.Itoa(pid), "/f")
 	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("taskkill: %v", err)
 	}
 	return nil
 }
+
+func matchProcess(appname string, processname string) bool {
+	// on Windows the process name will include the extension
+	return appname == processname+".exe"
```

Comment:
On Windows, doesn't `os.Args[0]` include the `.exe`?

Reply:
This was bothering me also, so 
After a bit of investigation

- `os.Args[0]` returns whatever was typed to launch caddy so

> caddy start

os.Args[0] will return `caddy` but

> caddy.exe start

will return `caddy.exe`

and to further make fun `get_process` in powershell returns `caddy` no matter what and tasklist on cmd window will return `caddy.exe` no matter what.

I think I will expand the windows function to match process name with or without `.exe`.

Also I noticed I have swapped around the appname and processname to make it even more confusing.

Reply:
Oh lovely! So I guess we need to standardize the name?

Wait, what if we use `taskkill /im` instead?

> **/im** Specifies the image name of the process to be terminated. Use the wildcard character (\*) to specify all image names.

I don't really know what an "image name" is, but maybe it just means the process name. Could we experiment with that?

Reply:
The danger there is that `taskkill /im caddy*` will kill `caddy` `caddy.exe` or even `caddyshack.exe` so that's not really possible.  I'll push another change that should do the trick.

Reply:
Well what about `taskkill /im os.Args[0]`?

Reply:
no because they are two seperate processes and 1 may have started as `caddy` and the other started as `caddy.exe`

Reply:
I've pushed a change now which consistently returns the appname with .exe

Reply:
The process list provided by `go-ps` always return process names with `.exe` so this will work consistently to find the process.